providers/pool.rb

```diff
@@ -13,7 +13,17 @@ def whyrun_supported?
 
 action :create do
   if @current_resource.exists
-    Chef::Log.info "#{@new_resource} already exists - nothing to do."
+    if @current_resource.pg_num == @new_resource.pg_num
```

Comment:
I can be quite dumb... but why not testing `@current_resource.pgp_num == @new_resource.pgp_num` also? :)